doc/api/tls.md

```diff
@@ -2,90 +2,139 @@
+The TLS/SSL is a public/private key infrastructure (PKI). Each client and
+server must have a *private key*.
+
+Private keys can be generated in multiple ways. The example below illustrates
+use of the OpenSSL command-line interface to generate a 2048-bit RSA private
+key:
 
 ```
 openssl genrsa -out ryans-key.pem 2048
 ```
 
-All servers and some clients need to have a certificate. Certificates are public
-keys signed by a Certificate Authority or self-signed. The first step to
-getting a certificate is to create a "Certificate Signing Request" (CSR)
-file. This is done with:
+With TLS/SSL, all servers (and some clients) must have a *certificate*.
+Certificates are *public keys* that correspond to a private key, and that are
+digitally signed either by a Certificate Authority or by the owner of the
+private-key (such certificates are referred to as "self-signed"). The first
+step to obtaining a certificate is to create a *Certificate Signing Request*
+(CSR) file.
+
+The OpenSSL command-line interface can be used to generate a CSR for a private
+key:
 
 ```
 openssl req -new -sha256 -key ryans-key.pem -out ryans-csr.pem
 ```
 
-To create a self-signed certificate with the CSR, do this:
+Once the CSR file is generated, it can either be sent to a Certificate
+Authority for signing or used to generate a self-signed certificate.
+
+Creating a self-signed certificate using the OpenSSL command-line interface
+is illustrated in the example below:
 
 ```
 openssl x509 -req -in ryans-csr.pem -signkey ryans-key.pem -out ryans-cert.pem
 ```
 
-Alternatively you can send the CSR to a Certificate Authority for signing.
+### Perfect Forward Secrecy
+
+<!-- type=misc -->
+
+The term "[Forward Secrecy]" or "Perfect Forward Secrecy" describes a feature of
+key-agreement (i.e., key-exchange) methods. That is, the server and client keys
+are used to negotiate new keys that are used specifically and only for the
+current communication session. Practically, this means that even if a private
+key is compromised, communication can only be decrypted by eavesdroppers if the
+attacker manages to obtain the key-pair specifically generated for the session.
+
+Perfect Forward Secrecy is achieved by randomly generating a key pair for
+key-agreement on every TLS/SSL handshake (in contrast to using the same key for
+all sessions). Methods implementing this technique are called "ephemeral".
+
+Currently two methods are commonly used to achieve Perfect Forward Secrecy (note
+the character "E" appended to the traditional abbreviations):
+
+* [DHE] - An ephemeral version of the Diffie Hellman key-agreement protocol.
+* [ECDHE] - An ephemeral version of the Elliptic Curve Diffie Hellman
+  key-agreement protocol.
 
-For Perfect Forward Secrecy, it is required to generate Diffie-Hellman
-parameters:
+Ephemeral methods may have some performance drawbacks, because key generation
+is expensive.
+
+To use Perfect Forward Secrecy using `DHE` with the `tls` module, it is required
```

Comment:
Looks great! Let's say something about `ECDHE`, though. Like `ECDHE` does not require any configuration except `ecdheCurve` in server's option which is provided by default.